app.rb

```diff
@@ -1,7 +1,43 @@
 require 'sinatra/base'
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+  enable :sessions
+  get '/' do
+    erb :homepage
+  end
+
+  post '/log_name' do
+    session[:player_name] = params[:player_name]
+    redirect '/play'
+  end
+
+  get '/play' do
+    @player_name = session[:player_name]
+    erb :play
+  end
+
+  post '/selection' do
+    @player_weapon = params[:weapon]
+    @computer_weapon = ["Rock", "Paper", "Scissors"].sample
+    @result = battle(@player_weapon, @computer_weapon)
+    erb :selection
+  end
+
+  def battle(player_weapon, computer_weapon)
```

Comment:
Nice that you started by creating a method on app.rb

The next step would be to move this into another class and then use that class in app.rb